ecl/eclagent/eclagent.cpp

```diff
@@ -2502,6 +2502,8 @@ bool EclAgent::checkPersistUptoDate(IRuntimeWorkflowItem & item, const char * lo
         errText.appendf("Rebuilding PERSIST('%s'): Saved CRC isn't present", logicalName);
```

Comment:
```
 if (!isResult(lfn, (unsigned)-2))
         errText.appendf("Building PERSIST('%s'): It hasn't been calculated before", logicalName);
 else if (!isResult(crcName, (unsigned)-2))
         errText.appendf("Rebuilding PERSIST('%s'): Saved CRC isn't present", logicalName);
 else if (isFile && !fileExists(logicalName))
         errText.appendf("Rebuilding PERSIST('%s'): Persistent file does not exist", logicalName);
 else if (!item.queryPersistRefresh())
        return true;
 else
```
So for REFRESH(false) you should still need to check that the file actually exists i.e. ```else if (isFile && !fileExists(logicalName))``` should still precede  ```else if (!item.queryPersistRefresh())``` (as it does) but I don't think you want to rebuild if the crc isn't present and the file is found by logicalname and actually exists on disk. As you pointed out, the crc is wanted so as to check that it is the same as the one requested, if not rebuild i.e. refresh which seems to me exactly what REFRESH(false) is asking it not to do, no? So a direct swap, as I previously suggested, isn't complete - I think you want to move
```
 else if (!isResult(crcName, (unsigned)-2))
         errText.appendf("Rebuilding PERSIST('%s'): Saved CRC isn't present", logicalName);
```
to after the refresh conditional, i.e. to the last elseif before the else. I think I'm saying that the order looked incorrect in the first place.